File: akilli_beslenme_mobil/test_overpass.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  String type = 'gym';
  double lat = 41.0082;
  double lng = 28.9784;

  String query = '''
    [out:json];
    (
      node["leisure"="fitness_centre"](around:5000, $lat, $lng);
      way["leisure"="fitness_centre"](around:5000, $lat, $lng);
    );
    out center;
  ''';

  final String url = 'https://overpass-api.de/api/interpreter';

  try {
    final response = await http.post(
      Uri.parse(url),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
        'User-Agent': 'AkilliBeslenmeApp/1.0',
        'Accept': '*/*'
      },
      body: 'data=${Uri.encodeComponent(query)}',
    );
    
    print("Status: ${response.statusCode}");
    if (response.statusCode == 200) {
      final data = json.decode(response.body);
      final List elements = data['elements'] ?? [];
      print("Elements count: ${elements.length}");
    } else {
      print("Body: ${response.body}");
    }
  } catch(e) {
    print("Error: $e");
  }
}

File: test_overpass.js
const https = require('https');

const query = `[out:json];
(
  node["leisure"="fitness_centre"](around:5000, 41.0082, 28.9784);
);
out center;`;

const options = {
  hostname: 'overpass-api.de',
  port: 443,
  path: '/api/interpreter',
  method: 'POST',
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded',
    'User-Agent': 'AkilliBeslenmeApp/1.0'
  }
};

const req = https.request(options, (res) => {
  console.log(`STATUS: ${res.statusCode}`);
  res.setEncoding('utf8');
  let data = '';
  res.on('data', (chunk) => {
    data += chunk;
  });
  res.on('end', () => {
    console.log(`BODY: ${data.substring(0, 200)}`);
  });
});

req.on('error', (e) => {
  console.error(`problem with request: ${e.message}`);
});

req.write('data=' + encodeURIComponent(query));
req.end();
